api/src/main/java/com/graphhopper/util/DistanceCalcEarth.java

```diff
@@ -254,6 +305,10 @@ public boolean isCrossBoundary(double lon1, double lon2) {
         return abs(lon1 - lon2) > 300;
     }
 
+    protected boolean hasElevationDiff(double a, double b) {
+        return a != b && !Double.isNaN(a) && !Double.isNaN(b);
+    }
```

Comment:
Why do we need this method? If calcDist3D is called then the elevation values should be valid and not NaN.

Reply:
I thought that any voids in elevation data show up as NaN?